src/flint/fq_nmod_mat.jl

```diff
@@ -700,34 +686,6 @@ function (a::fqPolyRepMatrixSpace)()
   return z
 end
 
-function (a::fqPolyRepMatrixSpace)(b::Integer)
-   M = a()
-   for i = 1:nrows(a)
-      for j = 1:ncols(a)
-         if i != j
-            M[i, j] = zero(base_ring(a))
-         else
-            M[i, j] = base_ring(a)(b)
-         end
-      end
-   end
-   return M
-end
-
-function (a::fqPolyRepMatrixSpace)(b::ZZRingElem)
-   M = a()
-   for i = 1:nrows(a)
-      for j = 1:ncols(a)
-         if i != j
-            M[i, j] = zero(base_ring(a))
-         else
-            M[i, j] = base_ring(a)(b)
-         end
-      end
-   end
-   return M
-end
-
 function (a::fqPolyRepMatrixSpace)(b::fqPolyRepFieldElem)
```

Comment:
I would expect this one and the ones below to work generically

Reply:
While there is a generic method, this method is actually optimized (though I don't know how effective the optimization is). I am reluctant to remove it as part of this PR. If someone would like to thoroughly test it and compare the performance (and allocations) of this code to the generic one, and then possibly (depending on the outcome) remove this here: fine by me. But I didn't want this PR to get bogged down by this kind of work (it's already been in the works for over a year at this point!)